src/main/java/bit/javaoop/AbstractEmployee.java

```diff
@@ -1,29 +0,0 @@
-package bit.javaoop;
-
-public abstract class AbstractEmployee {
```

Comment:
Ej! :) Nie chcieliśmy, żebyście usuwali klasy z zajęć, żeby wymusić na was wprowadzenie wyższej abstrakcji.

Klasa abstrakcyjna wprowadza obowiązek zaimplementowania swoich abstrakcyjnych metod wśród wszystkich swoich dzieci. Jak może w przypadku klasy Employee nie było to jakoś mega obowiązkowe (bo można wprawdzie przyjąć, że `Employee` to klasa bazowa każdego pracownika) tak już w przypadku klasy Manager, było. 
Po pierwsze: zauważ, że każdy Manager ma inną politykę zatrudniania pracowników. Chcemy zatem wymusić na nich by ją "po swojemu" zaimplementowali. Tworzymy więc abstrakcyjną metodę `canHire()`. Dzięki temu szczegółowi Managerzy dziedzicząc po niej muszą ją zaimplementować. Taka kontrola kodu.
Po drugie: Twoja obecna klasa Manager nie posiada metody `canHire` co oznacza, że tworząc listę managerów o typie `Manager` nie możemy sprawdzić, czy któryś manager z tej listy może kogoś zatrudnić. Tj. nie możemy zastosować tutaj pełni funkcjonalności **polimorfizmu**, a to chcieliśmy głównie osiągnąć w tym zadanku.

Chciałem coś jeszcze napisać, ale straciłem wątek xd Jak będziesz miał czas spróbuj sobie na szybko to przepisać z klasami abstrakcyjnymi. A na zajęciach spróbuje wyjaśnić na przykładzie o co mi z tym wszystkim chodziło.

Reply:
Dzięki za obszerne review. Zmieniłem klasę Manager na abstrakcyjną, czy mógłbyś zerknąć czy o to mniej więcej chodziło? ;)

Reply:
Jasne :)